frontend/src/pages/agreements/details/AgreementBudgetLines.js

```diff
@@ -20,6 +20,23 @@ export const AgreementBudgetLines = ({ agreement, isEditMode, setIsEditMode }) =
     const navigate = useNavigate();
     const isGlobalAlertActive = useSelector((state) => state.alert.isActive);
 
+    // Checks for who can edit budget lines
+    const loggedInUserId = useSelector((state) => state?.auth?.activeUser?.id);
+    const isUserAgreementCreator = agreement?.created_by === loggedInUserId;
+    const isUserTheProjectOfficer = agreement?.project_officer === loggedInUserId;
+    const isUserOnAgreementTeam = agreement?.team_members?.some((member) => member.id === loggedInUserId);
+    const isUserCreatorOfAnyBudgetLines = agreement?.budget_line_items?.some(
+        (bli) => bli.created_by === loggedInUserId
+    );
+    // TODO: add check if user is on the Budget Team
```

Comment:
no way of checking budget team membership as far as I know

Reply:
I think we'd expect them to have the "BudgetTeam" role within their JWT role: [] property. This is new in my pending PR, so should be able to address this _soon_